tests/virt_autotest/login_console.pm

```diff
@@ -120,6 +120,12 @@ sub login_to_console {
             #send key 'up' to stop grub timer counting down, to be more robust to select xen
             send_key 'up';
             save_screenshot;
+            #If sol console is stuck, reconnect it and resend key up
+            if (check_screen 'virttest-bootmenu-xen-kernel-count-down', 1) {
+                reset_consoles;
+                select_console 'sol', await_console => 0;
+                send_key 'up';
+            }
```

Comment:
I do not think the new needle can cover suitable amount of qualified scenarios for you to reset console subsequently.  Maybe the new needle is not necessary at all.  

For example, the following code snippet can also serve your purpose:

```
if (is_xen_host) {
    #send key 'up' to stop grub timer counting down, to be more robust to select xen
            
    for (1 .. 20) {
        disable_grub_timeout_sol;
        select_grub_xen_sol; 
    }
}

sub disable_grub_timeout_sol {
    reset_consoles;
    select_console 'sol', await_console => 0;
    for (1 .. 5) {
        send_key 'up';
        save_screenshot;
    }
}

sub select_grub_xen_sol {
    reset_consoles;
    select_console 'sol', await_console => 0;
    for (1 .. 20) {
        send_key 'down';
        last if check_screen 'virttest-bootmenu-xen-kernel', 5;
    }
```